### PR TITLE
Create anonymous-policy and token from non-default partitions.

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Ensure the anonymous token in the default partition is updated to support cross-partition DNS resolution even if the server-acl-init is running in a non-default partition. This ensures DNS lookups work even if the default partition is on VMs and hence does not have the requisite policy required to do the same.
- Update SDK to the latest version that update which updates ACL.Tokens.Master to ACL.Tokens.InitialManagement

How I've tested this PR:
- Acceptance tests don't break
- Tested on HCP with help from the cloud team.

How I expect reviewers to test this PR:
- 👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by ndhanushkodi — APPROVED** [review]
Looks great Ashwin! To make sure I understand correctly-- from any new partition where server-acl-init is run, it will idempotently configure the anonymous policy in the default partition?

**Comment by ndhanushkodi** [comment]
Oh I see it got updated to draft, just re-request me if you need a review again later

**Comment by thisisnotashwin** [comment]
> Looks great Ashwin! To make sure I understand correctly-- from any new partition where server-acl-init is run, it will idempotently configure the anonymous policy in the default partition?

Exactly!!  

> Oh I see it got updated to draft, just re-request me if you need a review again later

Yeah. Iryna and I had a conversation and I think I know how I want to test this. Will un-draft it once I have a unit test and re-request a review. Sorry bout that.

File: control-plane/go.mod [hunk 1]
```diff
@@ -129,4 +129,6 @@ require (
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
+replace github.com/hashicorp/consul/sdk v0.9.0 => github.com/hashicorp/consul/sdk v0.4.1-0.20220120214936-7568f3a102a8
```
Comment (thisisnotashwin) [inline control-plane/go.mod:132]: This was to update the version of the `consul/sdk` without impacting the version of the `consul/api`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -70,7 +70,7 @@ commands:
         type: string
       consul-k8s-image:
         type: string
-        default: "docker.mirror.hashicorp.services/hashicorpdev/consul-k8s-control-plane:latest"
+        default: "ashwinvenkatesh/consul-k8s@sha256:9b46fae446fcf99c715d9063231857eef5525830f2cd3cdcb46266c84db11deb"
```
Comment (thisisnotashwin) [inline .circleci/config.yml:73]: Will be deleted before merge.

File: control-plane/subcommand/server-acl-init/command_ent_test.go [hunk 3]
```diff
@@ -206,6 +208,54 @@ func TestRun_ConnectInject_NamespaceMirroring(t *testing.T) {
 	}
 }
 
+// Test that the anonymous token is created in the default partition from
```
Comment (ishustava) [inline control-plane/subcommand/server-acl-init/command_ent_test.go:211]:
```suggestion
// Test that the anonymous token policy is created in the default partition from
```

**Review by ishustava — APPROVED** [review]
Looks great! Just a couple of minor edits

**Review by ndhanushkodi — APPROVED** [review]
Looks great!! The test is :chefs_kiss!!

**Pull request merged**